src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CompoundAssignmentOperator.cs

```diff
@@ -274,7 +274,16 @@ BoundExpression rewriteAssignment(BoundExpression leftRead, BoundExpression righ
             Debug.Assert(rewrittenReceiver.Type is { });
             var variableRepresentsLocation = rewrittenReceiver.Type.IsValueType || rewrittenReceiver.Type.Kind == SymbolKind.TypeParameter;
 
-            var receiverTemp = _factory.StoreToTemp(rewrittenReceiver, out assignmentToTemp, refKind: variableRepresentsLocation ? RefKind.Ref : RefKind.None);
+            var receiverTemp = _factory.StoreToTemp(
+                rewrittenReceiver,
+                out assignmentToTemp,
+                refKind: variableRepresentsLocation ? RefKind.Ref : RefKind.None,
+                isKnownToReferToTempIfReferenceType: !variableRepresentsLocation || rewrittenReceiver.Type.IsValueType ||
```

Comment:
>!variableRepresentsLocation || rewrittenReceiver.Type.IsValueType || [](http://example.com/codeflow?start=53&length=68)

Could this be simplified to `!rewrittenReceiver.Type.IsReferenceType || !HasHome(...)`?